src/ir/var.rs

```diff
@@ -130,6 +131,71 @@ fn default_macro_constant_type(value: i64) -> IntKind {
     }
 }
 
+/// Determines whether a set of tokens from a CXCursor_MacroDefinition
+/// represent a function-like macro. If so, calls the func_macro callback
+/// and returns `Err(ParseError::Continue)` to signal to skip further
+/// processing. If conversion to UTF-8 fails (it is performed only where it
+/// should be infallible), then `Err(ParseError::Continue)` is returned as well.
+fn handle_function_macro(
+    cursor: &clang::Cursor,
+    tokens: &[ClangToken],
+    callbacks: &dyn crate::callbacks::ParseCallbacks,
+) -> Result<(), ParseError> {
+    fn is_abutting(a: &ClangToken, b: &ClangToken) -> bool {
+        unsafe {
+            clang_sys::clang_equalLocations(
+                clang_sys::clang_getRangeEnd(a.extent),
+                clang_sys::clang_getRangeStart(b.extent),
+            ) != 0
+        }
+    }
+
+    let is_functional_macro =
+        // If we have libclang >= 3.9, we can use `is_macro_function_like()` and
+        // avoid checking for abutting tokens ourselves.
+        cursor.is_macro_function_like().unwrap_or_else(|| {
```

Comment:
Ah, very nice find! This will allow us to even avoid the extra allocation of two vectors off tokens when we've got rid of libclang 3.8.

Mind leaving a `TODO` comment here like:

```
// TODO: Once we don't need the fallback check here we can lift this check up and avoid
// allocating the `tokens` vector for non-functional macros.
```

Or something of that sort?

Reply:
I think you are talking about the fact that `handle_function_macro` takes a parameter that is a slice of tokens.

`handle_function_macro` still needs that slice of `tokens`, because it still has to convert them to their spellings in order to call the `func_macro` callback.

I think I do not understand. Can you clarify what `Vec`s we can get rid of someday ?

Reply:
Ah, never mind. I understand now. I will add the comment within a few hours.

Reply:
I think the comment I added conveys the right meaning.